components/server/src/workspace/workspace-starter.ts

```diff
@@ -332,6 +334,37 @@ export class WorkspaceStarter {
         }
     }
 
+    protected blockedRepositories: { urlRegExp: RegExp; blockUser: boolean }[] | undefined;
+    protected getBlockedRepositories() {
+        if (!this.blockedRepositories) {
+            this.blockedRepositories = [];
+            if (this.config.blockedRepositories) {
+                for (const { blockUser, urlRegExp } of this.config.blockedRepositories) {
+                    this.blockedRepositories.push({
+                        blockUser,
+                        urlRegExp: new RegExp(urlRegExp),
+                    });
+                }
+            }
+        }
+        return this.blockedRepositories;
+    }
+
+    protected async checkBlockedRepository(user: User, contextURL: string) {
+        const hit = this.getBlockedRepositories().find((r) => !!contextURL && r.urlRegExp.test(contextURL));
+        if (!hit) {
+            return;
+        }
+        if (hit.blockUser) {
+            try {
+                await this.userService.blockUser(user.id, true);
+            } catch (error) {
+                log.error("Failed to block user.", error);
```

Comment:
nit: `log.error({ userId: user.id, workspaceId: workspace.id }, "Failed to block user.", error, { contextURL });`